great_expectations/core/util.py

```diff
@@ -364,6 +364,27 @@ def ensure_json_serializable(data):
         )
 
 
+def safe_deep_copy(data, memo=None):
+    """
+    This method makes a copy of a dictionary, applying deep copy to attribute values, except for non-pickleable objects.
+    """
+    if isinstance(data, (pd.Series, pd.DataFrame)) or (
+        pyspark and isinstance(data, pyspark.sql.DataFrame)
+    ):
+        return data
+
+    if isinstance(data, (list, tuple)):
+        return [safe_deep_copy(data=element, memo=memo) for element in data]
+
+    if isinstance(data, dict):
+        return {
+            key: safe_deep_copy(data=value, memo=memo) for key, value in data.items()
+        }
+
+    # noinspection PyArgumentList
+    return copy.deepcopy(data, memo)
```

Comment:
Could we type annotate this please?

Reply:
@cdkini This follows the deepcopy API, so I did not want to alter it.